xmake/rules/c++/modules/modules_support/msvc.lua

```diff
@@ -23,49 +23,73 @@ import("core.tool.compiler")
 import("core.project.project")
 import("core.project.depend")
 import("core.project.config")
+import("core.base.hashset")
 import("utils.progress")
 import("private.action.build.object", {alias = "objectbuilder"})
 import("common")
 
 -- add a module or header unit into the mapper
 --
 -- e.g
--- /headerUnit:angle cstdint=cstdint.ifc
+-- /headerUnit:angle Foo=build/.gens/Foo/rules/modules/cache/Foo.ifc
 -- /headerUnit:angle glm/mat4x4.hpp=Users\arthu\AppData\Local\.xmake\packages\g\glm\0.9.9+8\91454f3ee0be416cb9c7452970a2300f\include\glm\mat4x4.hpp.ifc
 --
-function _add_module_to_mapper(argument, module)
-    local mapflags = common.localcache():get("mapflags") or {}
-    local mapflag = format("%s %s", argument, module)
-    if table.contains(mapflags, mapflag) then
+function _add_module_to_mapper(target, argument, name, bmifile, deps)
+    local modulemap = _get_modulemap_from_mapper(target)
+    local mapflag = {argument, name .. "=" .. bmifile}
+
+    if modulemap[name] then
         return
     end
-    table.insert(mapflags, mapflag)
-    common.localcache():set("mapflags", mapflags)
+
+    for _, t in ipairs(project:targets()) do
+        if not t:name() == target:name() then
+            local t_modulemap = _get_modulemap_from_mapper(t)
+            if t_modulemap[name] then
+                mapflag = t_modulemap[name].flag
+                break
+            end
+        end
+    end
```

Comment:
we can optimize it, remove for-loop 

and it should use `project.targets()` instead of `project:targets()`

```lua
if not project.targets()[target:name()] then
      local t_modulemap = _get_modulemap_from_mapper(t)
      if t_modulemap[name] then
           mapflag = t_modulemap[name].flag
      end
end
```